File: android/app/src/main/java/com/audiobridge/app/network/ProtocolConstants.kt
package com.audiobridge.app.network

object ProtocolConstants {
    const val MAGIC: UShort = 0xCDABu
    const val HEADER_SIZE = 18
    const val MAX_PAYLOAD_SIZE = 1200

    const val DEFAULT_DATA_PORT = 54322
    const val DEFAULT_CONTROL_PORT = 54321

    // Parametri riproduzione audio (48kHz stereo)
    const val SAMPLE_RATE = 48000
    const val CHANNELS = 2
    const val FRAME_SIZE_MS = 20
    const val FRAME_SAMPLES = SAMPLE_RATE * FRAME_SIZE_MS / 1000

    // Jitter buffer: ~60ms = 3 frame
    const val JITTER_BUFFER_FRAMES = 3

    // Keep-alive (same as desktop)
    const val KEEP_ALIVE_INTERVAL_MS = 3000L
    const val KEEP_ALIVE_TIMEOUT_MS = 10000L

    // mDNS service discovery (Android NsdManager requires trailing dot)
    const val SERVICE_TYPE_NSD = "_audiobridge._tcp."

    object Flags {
        const val KEYFRAME = 0x01
        const val SILENCE = 0x02
        const val CONFIG_CHANGE = 0x04
        const val PCM_RAW = 0x08
    }
}

File: desktop/AudioBridge.Desktop/Network/ProtocolConstants.cs
using System;

namespace AudioBridge.Desktop.Network;

public static class ProtocolConstants
{
    public const ushort Magic = 0xCDAB;
    public const int HeaderSize = 18;

    public const int DefaultControlPort = 54321;
    public const int DefaultDataPort = 54322;
    public const string ServiceType = "_audiobridge._tcp.local";

    public const int MaxPayloadSize = 1200;
    public const int Mtu = 1280;

    public const int KeepAliveIntervalMs = 3000;
    public const int KeepAliveTimeoutMs = 200;
    public const int MaxMissedPongs = 3;
    public const int KeepAliveAbsoluteTimeoutMs = 10000;

    public const int DefaultSampleRate = 48000;
    public const int DefaultChannels = 2;
    public const int DefaultBitrate = 256000;
    public const int DefaultFrameSizeMs = 20;

    [Flags]
    public enum PacketFlags : byte
    {
        None = 0,
        Keyframe = 1 << 0,
        Silence = 1 << 1,
        ConfigChange = 1 << 2,
        PcmRaw = 1 << 3,
    }

    public const int RawFrameMs = 5;
}
